04-cs-fundamentals/internship/binary-search-trees.md

```diff
@@ -1,23 +1,253 @@
-# Trees
+# Binary Search Trees
+
 ## Learning Goals
-Get familiar with Binary Search Trees and the following algorithms related to it
-+ Search
-+ Insert value
-+ Delete value
-+ Find height
-+ Depth first traversals: pre-order, in-order, post-order
-+ Breadth first traversal
-
-We'll be covering BSTs in two parts. In the first session, we'll cover the recursive approach. In a later session, we'll cover the iterative approach.
-
-## Required
-Revise what you've already learned in the CS Fundamentals classroom curriculum:
-+ [Understanding function calls](https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/04-cs-fundamentals/classroom/Understanding%20function%20calls.md) and [Introduction to Recursion](https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/04-cs-fundamentals/classroom/Introduction%20to%20Recursion.md)
-+ Revise or author exercises related to recursion. See [Assignments and Additional Resources for Recursion](https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/04-cs-fundamentals/classroom/Introduction%20to%20Recursion.md#assignments)
-+ Before going through iterative approaches to BST algorithms, revise [Stacks and Queues](https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/04-cs-fundamentals/internship/Stacks%20and%20Queues.md)
 
+By the end of this lesson you should be able to:
+
+- Compare a Binary Tree to a Linked List
+- Explain how a Binary Search Tree differs from a generic Binary Tree
+- Write methods to perform the following on a Binary Search tree:
+  - Search
+  - Insert value
+  - Delete value
+  - Find height
+  - Perform traversals including: 
+    - Depth first traversals: pre-order, in-order, post-order
+    - Breadth first traversal
+
+## Video Lesson
+
+- [Video Lesson](https://adaacademy.hosted.panopto.com/Panopto/Pages/Viewer.aspx?id=d9746397-8a10-43be-b1cc-aaaf00720b31)
+- [Slide Deck](https://docs.google.com/presentation/d/1jE2bQDXkoCIJmqBgFXuripXftzrijuSY1JRKf2Ln7MY/edit#slide=id.p)
+
+
+## Limitations of Linked Lists
+
+Remember that finding an element in a Linked List requires a runtime of O(n).  This is because any search in a Linked List requires a linear search.  However once a Linked List node is found, removing or adding a node is O(1).  It would be ideal to combine the binary search O(log<sub>n</sub>) or a sorted array with the O(1) runtime to remove or add nodes.  This is what Binary Search Trees give us.
+
```

Comment:
Should this be `O(log(n))`? If we're talking about BSTs the base should be 2, but O(log<sub>n</sub>) implies the base is `n`.